aten/src/ATen/core/dispatch/DispatchKeyExtractor.h

```diff
@@ -151,7 +151,9 @@ struct TORCH_API DispatchKeyExtractor final {
         // no safe toTensorRef method, alas)
         ks = ks | ivalue.unsafeToTensorImpl()->key_set();
       } else if (C10_UNLIKELY(ivalue.isTensorList())) {
-        for (const at::Tensor& tensor : ivalue.toTensorList()) {
+        const c10::List<at::Tensor> tensorlist = ivalue.toTensorList();
+        for (const auto& tensor_ref : tensorlist) {
+          const at::Tensor& tensor = tensor_ref;
```

Comment:
I think this part could be changed in the same way as the ZeroTensorFallback.cpp change to avoid the error.